scripts/ci/strix_required_workflow_smoke.sh

```
@@ -0,0 +1,185 @@
+echo "Checking Strix workflow contract in $workflow_file"
+
+checkout_count="$(grep -Fc "uses: actions/checkout@" "$workflow_file" || true)"
+if [ "$checkout_count" != "1" ]; then
+	record_failure "Strix workflow must use actions/checkout exactly once for central trusted source checkout"
+fi
+
+assert_file_contains "$workflow_file" "Resolve trusted Strix source ref" "Strix workflow resolves central trusted source"
+assert_file_contains "$workflow_file" "workflow_repository" "Strix workflow reads required-workflow repository identity"
+assert_file_contains "$workflow_file" "workflow_sha" "Strix workflow prefers required-workflow source SHA"
+assert_file_contains "$workflow_file" "Checkout trusted Strix source" "Strix workflow checks out central source"
+assert_file_contains "$workflow_file" 'repository: ${{ steps.trusted_source.outputs.repository }}' "Strix workflow checks out resolved central repository"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)